src/ibex_bluesky_core/devices/simpledae/_reducers.py

```diff
@@ -429,3 +403,109 @@ def additional_readable_signals(self, dae: "SimpleDae") -> list[Device]:
             self.mon_integrals,
             self.det_integrals,
         ]
+
+
+class DSpacingMappingReducer(Reducer, StandardReadable):
+    """A DAE Reducer which exposes an array of d-spacings at each scan point.
+
+    This reducer produces one dimensional arrays of d-spacing; these arrays are generated
+    """
+
+    def __init__(
+        self,
+        *,
+        prefix: str,
+        detectors: npt.NDArray[np.int64],
+        l_total: sc.Variable,
+        two_theta: sc.Variable,
+        dspacing_bin_edges: sc.Variable,
+    ) -> None:
+        """Init.
```

Comment:
I think this or the class docstring could link to something like https://www.isis.stfc.ac.uk/Pages/TOF-neutron-diffraction.aspx or braggs law + ToF and distance travelled to wavelength